### DISCUSSION TITLE
Ability to Sort by Column Headers in the HyperSwitch Portal

### DISCUSSION THREAD

**Discussion by nara-golfdistrict-com** [juspay/hyperswitch#3881] [Ideas / Feature Requests]
Hello,

To quickly find the entries in table, we need the ability to sort by column headers in the HyperSwitch portal. Below is an example of a screen where this feature is not available. Ideally we would like this to be available where there is tabular data.

![image](https://github.com/juspay/hyperswitch/assets/145275696/bf6d2563-ad8b-4ccb-a8e6-8310191c85c1)

**Comment by AmanMaurya-HS** [comment]
Hey Nara, I agree that sorting is a very common feature that a table should have, but given we had filters and search w.r.t to each columns in the table we never had a strong reason to add the sorting feature. 

Are there any use cases/scenario you have faced wherein you felt sorting would be more helpful than searching or filtering?

**Reply by nara-golfdistrict-com** [reply]
@AmanMaurya-HS See below screenshot where I am trying to filter with no results. Currently your filter works only on specific columns. In-order for me to filter, first I have to go to the business profile section, grab the business profile ID and then come and search, which is very time consuming.
Sorting is very helpful as we have to deal with many payment processors. If the search worked then I can manage without sorting for a time being.

<img width="1236" alt="image" src="https://github.com/juspay/hyperswitch/assets/145275696/0675534e-cc76-4bd0-ac90-d9f418eb6dba">

![image](https://github.com/juspay/hyperswitch/assets/145275696/3db8f128-787b-4a64-92c4-f39a405dbacf)

**Comment by AmanMaurya-HS** [comment]
Understood your situation. We are currently in the process of enhancing the filters & search experience, once live hopefully you won't face this problem again. Until then let me see what we can do to quickly to improve yours and others experience. Thanks for the feedback Nara.
